Finessing configurator submission code

dpaks opened the pull request:
Lb @pkharat and VPN @SureshDH  yet to be merged to this branch.

dpaks committed:
code restructure

dpaks committed:
code reorg in base driver

dpaks committed:
refinements

pkharat committed:
Addressing review comments for LB agent and drivers code submission

dpaks committed:
Merge branch 'mitaka_nfp_13_july_2016' of https://github.com/oneconvergence/group-based-policy into finessing-configurator-submission-code

dpaks committed:
merged submitted code

dpaks committed:
class comments included inside the class

pkharat committed:
Merge branch 'lb-patch-submission-comments-addressed' of https://github.com/oneconvergence/group-based-policy into finessing-loadbalancer-submission-code

dpaks committed:
added nfp svc agent unit test

pkharat committed:
Synced OC git repo LB agent and driver code with gerrit LB agent and driver code

dpaks committed:
added nfp service driver and agent

pkharat committed:
Synced gerrit LB unit test code with mitaka_21_march_base branch

dpaks committed:
removed generailzed REST API from base driver

dpaks committed:
removed generailzed REST API from base driver

dpaks committed:
removed generailzed REST API from base driver

dpaks committed:
removed generailzed REST API from base driver

dpaks committed:
fixed all the testcases and pep8 issues

dpaks committed:
Merge branch 'mitaka_nfp_13_july_2016' of https://github.com/oneconvergence/group-based-policy into finessing-configurator-submission-code

dpaks committed:
removed assertRaises test cases

sdharav committed:
addressed the pep8 and code coverage

sdharav committed:
addressed pep8 isses

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:
```diff
@@ -192,51 +183,56 @@ def _test_handle_event(self, ev):
+                                ev, max_times=const.INITIAL_HM_RETRIES)
+                elif periodicity == const.FOREVER:
                     mock_hm_poll_event.assert_called_with(ev)
-            elif ev.id == 'CLEAR_HEALTHMONITOR':
```
Why is clear healthmonitor completely removed?

dpaks replied:
This was removed as it was just a placeholder. Anyways, I have written a test case for this.

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:
```diff
@@ -331,22 +326,10 @@ def test_configure_hm_forever_genericconfigeventhandler(self):
-    def test_clear_hm_genericconfigeventhandler(self):
-        """ Implements test case for clear health monitor method
-        of generic config event handler.
-
```
Why is this removed?

dpaks replied:
This was removed as it was just a placeholder with the child function having no definition. Anyways, I have written a test case for this.

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:
```diff
@@ -356,9 +339,5 @@ def test_handle_configure_healthmonitor_genericconfigeventhandler(self):
+        ev.id = const.EVENT_CONFIGURE_HEALTHMONITOR
         self._test_handle_periodic_event(ev)
-
-
```
Is this not needed because we moved to neutron base unit test?

dpaks replied:
This is unnecessary for run_tests script. We do not use unittest but basetest from neutron tests.

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py:
```diff
@@ -0,0 +1,159 @@
+        arg_dict = {'context': self.fo.context,
+                    'resource_data': self.fo.kwargs}
+        with mock.patch.object(
+                    sc, 'new_event', return_value='foo') as mock_sc_event, \
```
Lets get rid of "\" here.

dpaks replied:
### Addressed.

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py:
```diff
@@ -0,0 +1,159 @@
+    @mock.patch(__name__ + '.fo.FakeObjects.rpcmgr')
+    @mock.patch(__name__ + '.fo.FakeObjects.drivers')
+    @mock.patch(__name__ + '.fo.FakeObjects.sc')
+    def _get_NfpServiceEventHandler_object(self, sc, drivers, rpcmgr):
```
Can we name this function without uppercase chars. Follow the same convention as other function names.

dpaks replied:
### Addressed.

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py:
```diff
@@ -0,0 +1,159 @@
+                driver, 'run_heat', return_value=result) as mock_config_inte, (
+             mock.patch.object(
+                agent, '_get_driver', return_value=driver)):
+
```
Avoid too many spaces like this in the code. Not sure if PEP will complain, but keep one line gap which should be enough.

dpaks replied:
Jagadish, there is only one line gap.

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/drivers/firewall/test_vyos_fw_driver.py:
```diff
@@ -40,25 +37,75 @@ def __init__(self, *args, **kwargs):
+
     def test_configure_interfaces(self):
-        """ Implements test case for configure interfaces method
+        """ Implements test case for configure log forward method
```
Correct the comment. It is configuration of interfaces.

dpaks replied:
### Addressed.

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py:
```diff
@@ -265,8 +285,8 @@ def _fake_resource_data(self):
                     'stitching_mac': '00:0a:95:9d:68:16',
-                    'provider_interface_index': 'provider_interface_index',
-                    'stitching_interface_index': 'stitching_interface_index',
+                    'provider_interface_index': '2',
```
Are these values taken just for example? We removed the hardcoded valued form orchestrator right?

dpaks replied:
We never removed it from configurator side. We have to remove these from driver as well as schema files.

njagadish commented on gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py:
```diff
@@ -344,3 +364,8 @@ def __init__(self):
         self.id = 'dummy'
+
+
+class FakeEventGetNotifications(object):
```
Lets add comment for this class

dpaks replied:
### Addressed.

njagadish commented on gbpservice/nfp/configurator/Dockerfile:
```diff
@@ -47,21 +47,14 @@ RUN git clone -b stable/mitaka --single-branch https://github.com/openstack/neut
 RUN cp /group-based-policy/gbpservice/nfp/bin/nfp_configurator.ini /etc/
 RUN sed -i "s/log_forward_ip_address=*.*/log_forward_ip_address=VIS_VM_IP_ADDRESS/" /etc/nfp_configurator.ini
 RUN cp /group-based-policy/gbpservice/nfp/bin/policy.json /etc/
-RUN cp -r /group-based-policy/gbpservice/nfp/configurator/config /etc/nfp_config
```
Why is this not needed now?

dpaks replied:
This is not needed because this dir is used only by ASAv which we are not going to submit. Hence, we though of removing this empty directory for submission.

njagadish commented on gbpservice/nfp/configurator/drivers/firewall/vyos/vyos_fw_driver.py:
```diff
@@ -565,43 +500,28 @@ def create_firewall(self, context, firewall, host):
+            return common_const.STATUS_ACTIVE
+
+        err_msg += (("Reason: %r, Response Content: %r" %
+                     (resp.pop('message'), resp))
```
If there is a possibility of resp becoming None, there will be an exception. Please check the possibility.

dpaks replied:
Resp is constructed in the RestApi class. It can either be a string or a dict. I don't think that this will be an issue. Checked it.

pkharat committed:
Addressed review comments
1) Used '(' instead of '\'
2) Used neutron.tests.base class instead of unittest

pkharat committed:
Addressed review comments given by Jagadish

sdharav committed:
addressed review comments

sdharav committed:
addressed pep8 isses

dpaks committed:
added hm test cases, addressed review comments

dpaks committed:
fixed pep8

njagadish committed:
Merge pull request #510 from oneconvergence/finessing-configurator-submission-code-vpn-changes

Finessing configurator submission code vpn changes

njagadish committed:
Merge pull request #506 from oneconvergence/finessing-loadbalancer-submission-code

Finessing loadbalancer submission code

njagadish committed:
Merge pull request #511 from oneconvergence/mitaka-sync-gerrit-lb-unit-tests

Mitaka sync gerrit lb unit tests
